shared/health/src/main/java/io/pravega/shared/health/impl/HealthDaemonImpl.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.health.impl;
+
+import io.pravega.common.concurrent.ExecutorServiceHelpers;
+import io.pravega.shared.health.Health;
+import io.pravega.shared.health.HealthDaemon;
+import io.pravega.shared.health.HealthService;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Slf4j
+@RequiredArgsConstructor
+public class HealthDaemonImpl implements HealthDaemon {
+
+    /**
+     * The {@link Health} the daemon should report in the case it has not started querying the {@link HealthService}.
+     */
+    private static final Health INITIAL_HEALTH = Health.builder().build();
+
+    /**
+     * Represents the most recent {@link Health} information provided by the {@link ScheduledExecutorService}.
+     */
+    AtomicReference<Health> latest = new AtomicReference<Health>(INITIAL_HEALTH);
+
+    /**
+     * The {@link HealthService} associated with this {@link HealthDaemon}.
+     */
+    private final HealthService service;
+
+    /**
+     * The underlying {@link ScheduledExecutorService} used to executor the recurring service-level {@link Health} check.
+     */
+    private final ScheduledExecutorService executor = ExecutorServiceHelpers.newScheduledThreadPool(2, "health-daemon");
```

Comment:
1 thread should be enough.
make this low priority (there's an overload for that)